common/src/main/java/com/revenuecat/purchases/common/HTTPClient.kt

```diff
@@ -260,4 +286,26 @@ class HTTPClient(
         Store.AMAZON -> "amazon"
         else -> "android"
     }
+
+    private fun shouldSignResponse(endpoint: Endpoint): Boolean {
+        return endpoint.supportsSignatureValidation && entitlementVerificationMode.shouldVerify
+    }
+
+    private fun verifyResponse(
+        urlPath: String,
+        connection: URLConnection,
+        payload: String?,
+        nonce: String
+    ): HTTPResult.VerificationStatus {
+        return signingManagerIfEnabled?.verifyResponse(
+            requestPath = urlPath,
+            signature = connection.getHeaderField(HTTPResult.SIGNATURE_HEADER_NAME),
+            nonce = nonce,
+            body = payload,
+            requestTime = connection.getHeaderField(HTTPResult.REQUEST_TIME_HEADER_NAME),
+            eTag = getETagHeader(connection),
+        ) ?: HTTPResult.VerificationStatus.NOT_VERIFIED
```

Comment:
This coverage warning is probably important 😅

Reply:
Right, we should probably error out in this case since it should never happen... Basically it will only get here if `signingManagerIfEnabled` is null, but in that case we wouldn't get here since it means verification is disabled and this method is called only to verify. This also leads me to believe I should refactor this better to avoid this scenario... Will work on that refactor and ask for another review...

Reply:
Refactored things to avoid this issue in https://github.com/RevenueCat/purchases-android/pull/826